doc/protocols/scion-header.rst

```diff
@@ -787,19 +787,33 @@ Colibri Packet Timestamp
      0                   1                   2                   3
      0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |                             TsRel                             |
+    |                        Expiration Tick                        |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                TsRel / Original Payload Length                |
     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    |                             PckId                             |
+    |                         PckId / NOOP                          |
     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 
-Both fields ``TsRel`` and ``PckId`` contain arbitrary data when ``C=1``
+The field ``TsRel`` contains the ``Original Payload Length`` when ``R=1``.
```

Comment:
I think this only works if C=1, as then the TsRel and PckId fields are in principle unused/undefined.
But at least currently, for C=1, TsRel and PckId are used (in both directions, i.e., for R=0 and R=1) for the "single dispatcher workaround" when running a local topology.
(With the local topology, there is only one dispatcher for all the ASes, if not running the dockerized variant, leading to problems.)
See https://github.com/netsec-ethz/scion/pull/116
As you are proposing to overwrite those fields, do you have a suggestion on how to resolve those problems?

But more importantly, I don't think it is possible to overwrite TsRel with the Original Payload Length for R=1 in case of C=0.
The reason is that the Colibri timestamp (including TsRel) is needed for re-computing the HVF.
To compute the HVF when R=1, both TsRel and Original Payload Length are needed, so they need to be present as separate fields.

Reply:
Update: As discussed, the suggested change does not work for return traffic if C=0.